### PR TITLE
Allow null modifier in columns declaration for table functions

### PR DESCRIPTION
**Pull request by Avogar**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Allow null modifier in columns declaration for table functions

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Review by novikd — APPROVED** [review]

**Comment by qoega** [comment]
Can you check `generateRandom` also?

**Comment by Avogar** [comment]
> Can you check `generateRandom` also?

It also works now. These changes apply to all table functions

File: src/Parsers/ParserCreateQuery.h [hunk 1]
```diff
@@ -333,9 +333,21 @@ bool IParserColumnDeclaration<NameParser>::parseImpl(Pos & pos, ASTPtr & node, E
 
 class ParserColumnDeclarationList : public IParserBase
 {
+public:
+    explicit ParserColumnDeclarationList(bool require_type_ = true, bool allow_null_modifiers_ = false, bool check_keywords_after_name_ = false)
```
Comment (alexey-milovidov) [inline src/Parsers/ParserCreateQuery.h:337]: Why don't enable it everywhere unconditionally?

**Comment by qoega** [comment]
>It also works now. These changes apply to all table functions

Nice. We can add a test with it. Do we have some sort of schema fuzzer to check this table functions?

**Pull request merged**